packages/flutter/lib/src/material/material_state.dart

```diff
@@ -309,26 +301,32 @@ abstract class MaterialStateOutlineInputBorder extends OutlineInputBorder implem
   /// Creates a [MaterialStateOutlineInputBorder] from a [MaterialPropertyResolver<InputBorder>]
   /// callback function.
   ///
-  /// If used as a regular input border, the border resolved in the default state (the
-  /// empty set of states) will be used.
+  /// [MaterialStateOutlineInputBorder] should only be used with widgets
+  /// that document their support, like [InputDecoration.border]
+  /// (if used as a regular [InputBorder], it acts the same as
+  /// an empty `OutlineInputBorder()` constructor).
```

Comment:
In general I don't think this is always true, for this and documentation changes in this PR like this. If the resolve method resolves for no states, or `any`, then it will return not the default wherever used.

Reply:
I believe the current documentation is correct, since the resolve method is never called when it's used as a normal input border.

```dart
// Uses the border resolved in the default state.
final InputBorder border = _border.resolve(<WidgetState>{});
```

```dart
// No syntax error, but if it's a MaterialStateOutlineInputBorder
// then it's the same as an empty OutlineInputBorder() constructor.
final InputBorder border = _border;
```